firebase-vscode/src/data-connect/emulator-stream.ts

```diff
@@ -0,0 +1,133 @@
+import * as vscode from "vscode";
+import fetch from "node-fetch";
+import { Observable, of } from "rxjs";
+import { backOff } from "exponential-backoff";
+import { ResolvedDataConnectConfigs } from "./config";
+
+enum Kind {
+  KIND_UNSPECIFIED = "KIND_UNSPECIFIED",
+  SQL_CONNECTION = "SQL_CONNECTION",
+  SQL_MIGRATION = "SQL_MIGRATION",
+  VERTEX_AI = "VERTEX_AI",
+}
+enum Severity {
+  SEVERITY_UNSPECIFIED = "SEVERITY_UNSPECIFIED",
+  DEBUG = "DEBUG",
+  NOTICE = "NOTICE",
+  ALERT = "ALERT",
+}
+interface EmulatorIssue {
+  kind: Kind;
+  severity: Severity;
+  message: string;
+}
+
+type CompilerResponse = { result?: { issues?: EmulatorIssue[] } };
+
+export const emulatorOutputChannel =
+  vscode.window.createOutputChannel("Firebase Emulators");
+
+/**
+ *
+ * @param fdcEndpoint FDC Emulator endpoint
+ */
+export async function runEmulatorIssuesStream(
+  configs: ResolvedDataConnectConfigs,
+  fdcEndpoint: string,
+) {
+  const obsErrors = await getEmulatorIssuesStream(configs, fdcEndpoint);
+  const obsConverter = {
+    next(nextCompilerResponse: CompilerResponse) {
+      if (nextCompilerResponse.result && nextCompilerResponse.result.issues) {
+        for (const issue of nextCompilerResponse.result.issues) {
+          displayIssue(issue);
+        }
+      }
+    },
+    error(e: Error) {
+      console.log("Stream closed with: ", e);
+    },
+    complete() {
+      console.log("Stream Closed");
+    },
+  };
+  obsErrors.subscribe(obsConverter);
+}
+
+/**
+ * Based on the severity of the issue, either log, display notification, or display interactive popup to the user
+ */
+export function displayIssue(issue: EmulatorIssue) {
+  const issueMessage = `Data Connect Emulator: ${issue.kind.toString()} - ${issue.message}`;
+  if (issue.severity === Severity.ALERT) {
+    vscode.window.showErrorMessage(issueMessage);
+  } else if (issue.severity === Severity.DEBUG) {
```

Comment:
I guess this could be debated, but my interpretation of our severity levels was that `Severity.NOTICE` is the display notification level, and `Severity.DEBUG` is the log level.

Reply:
oh you're right, fixed